### ISSUE TITLE
Cannot convert _start_timezone

### ISSUE THREAD

**Issue by moozer** [ecederstrand/exchangelib#273]
Tweaking your example code, I fetch the calendar entries, using this:
```
...
    ids=account.calendar.filter(start__range=(
        EWSDateTime.from_datetime(dt_start),
        EWSDateTime.from_datetime(dt_end)
    )).only('subject', 'start') 
...
```

I get warnings like this
```
WARNING:exchangelib.fields:Cannot convert value 'Europe/Copenhagen' on field '_start_timezone' to type <class 'exchangelib.ewsdatetime.EWSTimeZone'> (unknown timezone ID)
WARNING:exchangelib.fields:Cannot convert value 'Europe/Paris' on field '_start_timezone' to type <class 'exchangelib.ewsdatetime.EWSTimeZone'> (unknown timezone ID)
```
and the resulting start time for my calendar entries are off by 2 hours.

I have no idea where to start debugging this.
Could I be missing some packages?

Some info:
INFO:exchangelib.version:API version "Exchange2013_SP1" worked but server reports version "V2_23". Using "Exchange2013_SP1"

➜ python --version
Python 2.7.13

➜ pip list | grep exchange
exchangelib (1.10.4)

**Comment by ecederstrand** [comment]
We were somehow unable to parse the timezone value on some incoming datetime fields. Can you capture the XML response as described in https://github.com/ecederstrand/exchangelib#troubleshooting and post it here? The interesting part is the `StartTimeZone` element so you can exclude the rest.

**Comment by moozer** [comment]
I have cut out an entire CalendarItem

```
              <t:CalendarItem>
                <t:ItemId Id="(some id)" ChangeKey="(some key)"/>
                <t:Subject>I do something at this hour</t:Subject>
                <t:Start>2017-10-24T08:00:00+02:00</t:Start>
                <t:StartTimeZone Name="" Id="Europe/Copenhagen">
                  <t:Periods>
                    <t:Period Bias="-PT1H" Name="Standard" Id="trule:Microsoft/Registry/Europe/Copenhagen/1-Standard"/>
                    <t:Period Bias="-PT2H" Name="Daylight" Id="trule:Microsoft/Registry/Europe/Copenhagen/1-Daylight"/>
                  </t:Periods>
                  <t:TransitionsGroups>
                    <t:TransitionsGroup Id="0">
                      <t:RecurringDayTransition>
                        <t:To Kind="Period">trule:Microsoft/Registry/Europe/Copenhagen/1-Daylight</t:To>
                        <t:TimeOffset>PT2H</t:TimeOffset>
                        <t:Month>3</t:Month>
                        <t:DayOfWeek>Sunday</t:DayOfWeek>
                        <t:Occurrence>-1</t:Occurrence>
                      </t:RecurringDayTransition>
                      <t:RecurringDayTransition>
                        <t:To Kind="Period">trule:Microsoft/Registry/Europe/Copenhagen/1-Standard</t:To>
                        <t:TimeOffset>PT3H</t:TimeOffset>
                        <t:Month>10</t:Month>
                        <t:DayOfWeek>Sunday</t:DayOfWeek>
                        <t:Occurrence>-1</t:Occurrence>
                      </t:RecurringDayTransition>
                    </t:TransitionsGroup>
                  </t:TransitionsGroups>
                  <t:Transitions>
                    <t:Transition>
                      <t:To Kind="Group">0</t:To>
                    </t:Transition>
                  </t:Transitions>
                </t:StartTimeZone>
              </t:CalendarItem>

```

**Comment by moozer** [comment]
Bonus info.
I have other entries with this timezone
```
                <t:StartTimeZone Name="(UTC+01:00) Brussels, Copenhagen, Madrid, Paris" Id="Romance Standard Time">

```
that works, and looking at `account.protocol.get_timezones()` I get
```
('Dateline Standard Time', '(UTC-12:00) International Date Line West')
('Samoa Standard Time', '(UTC+13:00) Samoa')
('UTC-11', '(UTC-11:00) Coordinated Universal Time-11')
...
('Azores Standard Time', '(UTC-01:00) Azores')
('Cape Verde Standard Time', '(UTC-01:00) Cape Verde Is.')
('Morocco Standard Time', '(UTC) Casablanca')
('UTC', '(UTC) Coordinated Universal Time')
('GMT Standard Time', '(UTC) Dublin, Edinburgh, Lisbon, London')
('Greenwich Standard Time', '(UTC) Monrovia, Reykjavik')
('W. Europe Standard Time', '(UTC+01:00) Amsterdam, Berlin, Bern, Rome, Stockholm, Vienna')
('Central Europe Standard Time', '(UTC+01:00) Belgrade, Bratislava, Budapest, Ljubljana, Prague')
('Romance Standard Time', '(UTC+01:00) Brussels, Copenhagen, Madrid, Paris')
('Central European Standard Time', '(UTC+01:00) Sarajevo, Skopje, Warsaw, Zagreb')
('W. Central Africa Standard Time', '(UTC+01:00) West Central Africa')
('Namibia Standard Time', '(UTC+01:00) Windhoek')
...
('UTC+13', '(UTC+13:00) Coordinated Universal Time+13')
('Tonga Standard Time', "(UTC+13:00) Nuku'alofa")
('Line Islands Standard Time', '(UTC+14:00) Kiritimati Island')
```
Which includes the timezone that works but not  `Europe/Copenhagen`

**Comment by ecederstrand** [comment]
Microsoft doesn't make it easy for us... I can find no mention of `Europe/Copenhagen` as a timezone ID in official docs (e.g. https://docs.microsoft.com/en-us/windows-hardware/manufacture/desktop/default-time-zones).

It's possible that `Microsoft/Registry/Europe/Copenhagen` is an adaption of the Olson timezone database, in which case we could use `pytz` to look up the correct timezone.

**Comment by ecederstrand** [comment]
Can you try https://github.com/ecederstrand/exchangelib/commit/89c19dbabff27da4ffaa37bafa20f065a9f97f9c ?

**Comment by moozer** [comment]
I did 
> pip3 uninstall exchangelib
> pip3 install git+https://github.com/ecederstrand/exchangelib.git -U

and now I don't see the warning anymore.

The returned datetime object/EWSDateTime are in UTC, but the time is correct. It would be nice if the timezone was the correct too.

A work around (for me):
```
    ids=account.calendar.filter(start__range=(...)).only('subject', 'start') 
    for i in ids:
        yield {'start': i.start.astimezone(tz), 'summary': i.subject}
```

**Comment by ecederstrand** [comment]
Thanks for the test!

The UTC datetimes are actually by design. Everything we fetch from the server is converted to UTC. This is because I think we should work with UTC datetimes internally, and that the presentation layer should handle the task of displaying datetimes in the local timezone as needed.

**Comment by ecederstrand** [comment]
Closing. Please feel free to reopen if you disagree :-)

**Comment by moozer** [comment]
That works for me too - clear design decision is key to success :-)

**Comment by moozer** [comment]
@utapyngo I tested it using the latest.
If you see my post from Oct 24, I write how I pip install it.

**Comment by utapyngo** [comment]
@moozer  pip cache does not work well with git, that's why our deploy scripts require all packages to be released to PyPI.

**Comment by ecederstrand** [comment]
Thanks for the heads-up. I'll release a new version tonight.

**Comment by ecederstrand** [comment]
Released v1.10.5 just now.

**Comment by mcv21** [comment]
Hi,

[sorry, if you'd rather a new issue, do say, but this seemed an obvious place to put this]

I'm not sure that presenting times in UTC is necessarily correct - most calendar events/invitations naturally have a timezone associated with them (and they're stored as `EWSDateTime` which is a tz-aware type). If I make an event in, say, BST, does it not make sense to store that with BST in the `tzinfo` field, rather than storing them in `UTC` (which is, in a sense, false) and keeping the original TZ in a separate field?

Relatedly, might it be worth flagging this a bit more explicitly in the docs? [e.g. that the tzinfo field does not reflect the event TZ, and that one should instead check `{_start,_end}_timezone` instead]
